### PR TITLE
BUG: make reference_level column::value separator dynamic based on too many vs too few separators

### LINKED ISSUE #119: error in ancombc error message
```bash
qiime composition ancombc \
    --i-table table.qza \
    --m-metadata-file metadata.tsv \
    --p-formula bodysite \
    --p-reference-levels bodysite:Tongue \
    --o-differentials dataloaf.qza
Plugin error from composition:

  Too many column-value pair separators found (`::`) in the following `reference_level`: "bodysite:Tongue"
```

This error message says "too many", but it should be "too few".

### PR DESCRIPTION
**Pull request by Oddant1**
Closes #119 by making the error message dynamic based on number of separators

### CODE REVIEW

**Review by lizgehret — COMMENTED** [review]
This is great, thanks @Oddant1! Will pull down locally to test, but this should be g2g.

File: q2_composition/_ancombc.py [hunk 1]
```diff
@@ -139,10 +139,19 @@ def _ancombc(table, metadata, formula, p_adj_method, prv_cut, lib_cut,
     for level in reference_levels:
         try:
             column, level_value = level.split('::')
-        except Exception:
-            raise ValueError('Too many column-value pair separators found'
+        except ValueError as e:
```
[the remark this answers is not in the document]
Comment (Oddant1) [reply q2_composition/_ancombc.py:142]: I'm not sure how I feel about this. In principle there's nothing wrong with them naming a value `:tongue`, but it does feel a bit silly. I think we can allow it though.

**Comment by Oddant1** [comment]
We floated the idea of what happens if their metadata actually contains `::tongue`. Then they would pass `bodysite::::tongue` and it would spit out `('bodysite', '', 'tongue')` which is not good. This makes it seem like we need to disallow `::` in metadata.

The idea was floated of using `.split('::', maxsplit=1)`, then we would get `('bodysite', '::tongue')`, but this raises another issue. If the value in their metadata was actually `bodysite::` and not `::tongue` they would still get `('bodysite', '::tongue')` when they would actually want `('bodysite::', 'tongue')`. It seems like we just need to disallow occurences of more than one `:` in a row in metadata.

**Comment by gregcaporaso** [comment]
@Oddant1, it is also ok for us just to fail with a descriptive error message. We've purposely tried to avoid putting restrictions on what is actually allowed in metadata, in favor of [these recommendations](https://docs.qiime2.org/2023.2/tutorials/metadata/#recommendations-for-identifiers). If a user is following the recommendations, none of these issues would come up, so we can always point them to the recommendations if one of these cases actually comes up in the wild. 

So I guess my vote would be to go with the `.split('::', maxsplit=1)`, and then provide a meaningful error message if we don't find either the column headers or the values. In that case you can note that colons in column headers or values can cause problems with this action.

**Comment by lizgehret** [comment]
@Oddant1 @gregcaporaso I just chatted with @ebolyen about this, and this would actually be a good place to utilize Collections (i.e. input would instead look like a dict of key:value pairs instead of having to list each column::value pair separately). In any case, this would be a breaking change so it's probably outside of the scope of this PR. I'll open a separate issue on this because this would probably the best actual solution.

File: q2_composition/_ancombc.py [hunk 2]
```diff
@@ -139,10 +139,19 @@ def _ancombc(table, metadata, formula, p_adj_method, prv_cut, lib_cut,
...
+            # More than one ::
+            if 'too many' in str(e):
```
Comment (gregcaporaso) [inline q2_composition/_ancombc.py:144]: Parsing Python's error messages feels risky to me as we don't have any guarantee (that I'm aware of) that those will be stable across versions. I would more explicitly check for the problems, something like the following (this code hasn't been tested so probably doesn't work as-is):

```python
fields = level.split('::', maxsplit=1)
if len(fields) == 1: 
    # raise your "too few" error

column, level_value = fields

if column not in md.columns:
    message = 'Column %s not in metadata.'
    if ':' in column:
        message = f"{message} This column name appears to contain a `:`, which can be a problem for this action."
...
```

**Review by gregcaporaso — CHANGES_REQUESTED** [review]

File: q2_composition/_ancombc.py (hunk 2 above)
Comment (Oddant1) [reply q2_composition/_ancombc.py:144]: I think you're right that they could just decide to change those error messages. I've addressed this and probably done enough to provide a reasonable stop gap until we can address the issue using a collection. Maybe @lizgehret and I can do that when this library stuff is done and she can start doing the Collections in usage examples and galaxy and all that.

**Review by lizgehret — APPROVED** [review]
these changes lgtm @Oddant1 - just pulled down locally to test, and these errors all make sense. thanks!

**Pull request merged**